closed/src/jdk.crypto.ec/share/classes/sun/security/ec/NativeECDHKeyAgreement.java

```diff
@@ -100,34 +102,43 @@ protected void engineInit(Key key, SecureRandom random)
                 ("Key must be an instance of PrivateKey");
         }
         /* attempt to translate the key if it is not an ECKey */
-        this.privateKey = (ECPrivateKeyImpl) ECKeyFactory.toECKey(key);
-        this.publicKey = null;
+        ECPrivateKey ecKey = (ECPrivateKey) ECKeyFactory.toECKey(key);
+        if (ecKey instanceof ECPrivateKeyImpl) {
+            this.privateKey = (ECPrivateKeyImpl) ecKey;
```

Comment:
The cast to `ECPrivateKey` is unnecessary; also suggest using pattern matching language feature:
```
        ECKey ecKey = ECKeyFactory.toECKey(key);
        if (ecKey instanceof ECPrivateKeyImpl keyImpl) {
            this.privateKey = keyImpl;
```